---
search:
    keywords: ['inheritance','继承','exception','异常']

---


# 类 Example A 继承 Exception，类 ExampleB 继承 Example A

# 有如下代码片断：

```
try{
    throw new ExampleB("b");
}catch（ExampleA e）{
    System.out.printfln（"ExampleA"）;
}catch（Exception e）{
    System.out.printfln（"Exception"）;
}
```

# 输出的内容应该是：

A.ExampleA     B.Exception     C.b     D.无

# 参考解答

输出为A。
类 ExampleB 继承 Example A,当抛出ExampleB时，会被ExampleA捕获到，所以会输出ExampleA

---


